test/e2e/network/service.go

```diff
@@ -683,6 +708,23 @@ func waitForApiserverUp(c clientset.Interface) error {
 	return fmt.Errorf("waiting for apiserver timed out")
 }
 
+// getEndpointNodesWithInternalIP returns a map of nodenames:internal-ip on which the
+// endpoints of the Service are running.
+func getEndpointNodesWithInternalIP(jig *e2eservice.TestJig) (map[string]string, error) {
```

Comment:
Let's move this function to jig.go (make it method of jig the same way as GetEndpointNodes is.

Reply:
Currently we already have a similar method, i.e. `jig.GetEndpointNodesWithIP(corev1.NodeInernalIP)`，it returns a map of node names -> list of internal IPs and would return empty list if no IP found.

I put the function `getEndpointNodesWithInternalIP` here because I think this function would only be used here, and it returns a map of node names -> ip(rather than a list of IP) and it would return an error if no IP found.